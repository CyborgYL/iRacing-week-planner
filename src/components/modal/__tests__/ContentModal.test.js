import { describe, test } from '@jest/globals';
import React from 'react';
import { shallow } from 'enzyme';

import ContentModal from '../ContentModal';

describe('components/modal/ContentModal', () => {
  test('renders correctly', () => {
    const save = jest.fn();
    const saveFavourites = jest.fn();
    const component = shallow(
      <ContentModal
        id="cars"
        title="Cars Modal"
        isOpen
        onClose={() => {}}
        save={save}
        content={[
          { id: 123, skuname: 'Ford', type: 'oval' },
          { id: 234, skuname: 'Toyota', type: 'oval' },
          { id: 456, name: 'Mazda', type: 'road' },
          { id: 645, name: 'McLaren', type: 'road' },
        ]}
        ownedContent={[123, 456, 645]}
        defaultContent={[456]}
        idField="id"
        typeFilter={{
          key: 'type',
          oval: 'oval',
          road: 'road',
        }}
        favourites={[456]}
        saveFavourites={saveFavourites}
      />,
    );

    expect(component).toMatchSnapshot();

    component.find({ id: 'cars-select-oval' }).props().onChange(true);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([123, 234, 456, 645]));
    save.mockClear();

    component.find({ id: 'cars-select-oval' }).props().onChange(false);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([456, 645]));
    save.mockClear();

    component.find({ id: 'cars-select-road' }).props().onChange(false);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([123, 456])); // 456 is default
    save.mockClear();

    component.find({ id: 'cars-select-road' }).props().onChange(true);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([123, 456, 645]));
    save.mockClear();

    component.find({ id: 'cars-select-all' }).props().onChange(true);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([123, 234, 456, 645]));
    save.mockClear();

    component.find({ id: 'cars-select-all' }).props().onChange(false);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([456])); // 456 is default
    save.mockClear();

    component.find({ id: 'cars-select-item-234' }).props().onChange(true);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([123, 234, 456, 645]));
    save.mockClear();

    component.find({ id: 'cars-select-item-123' }).props().onChange(false);
    expect(save).toBeCalledWith(expect.toIncludeSameMembers([456, 645]));
    save.mockClear();

    component.find({ id: 'cars-favourite-oval' }).props().onClick(true);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([123, 234, 456]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-oval' }).props().onClick(false);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([456]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-road' }).props().onClick(true);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([456, 645]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-road' }).props().onClick(false);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-all' }).props().onClick(true);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([123, 234, 456, 645]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-all' }).props().onClick(false);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-item-456' }).props().onClick(false);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([]));
    saveFavourites.mockClear();

    component.find({ id: 'cars-favourite-item-123' }).props().onClick(true);
    expect(saveFavourites).toBeCalledWith(expect.toIncludeSameMembers([123, 456]));
    saveFavourites.mockClear();
  });
});
