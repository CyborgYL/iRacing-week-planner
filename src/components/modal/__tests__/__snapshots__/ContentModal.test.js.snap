// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modal/ContentModal renders correctly 1`] = `
<Modal
  doneAction={[Function]}
  doneButtonText="Close"
  isOpen={true}
  onClose={[Function]}
  showFooter={true}
  title="Cars Modal"
>
  <div
    className="container-fluid"
  >
    <p>
      Use the checkbox to set the content you own, and use the star to set your favourite content.
    </p>
    <div
      className="row"
    >
      <div
        className="checkboxContainer col-sm-4"
      >
        <div>
          <Checkbox
            checked={false}
            disabled={false}
            id="cars-select-all"
            onChange={[Function]}
          >
            Select all
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={false}
            id="cars-favourite-all"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="checkboxContainer col-sm-4"
      >
        <div>
          <Checkbox
            checked={false}
            disabled={false}
            id="cars-select-oval"
            onChange={[Function]}
          >
            Select all oval
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={false}
            id="cars-favourite-oval"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="checkboxContainer col-sm-4"
      >
        <div>
          <Checkbox
            checked={true}
            disabled={false}
            id="cars-select-road"
            onChange={[Function]}
          >
            Select all road
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={false}
            id="cars-favourite-road"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
    <hr />
    <div
      className="row"
    >
      <div
        className="checkboxContainer col-md-6"
        key="123"
      >
        <div>
          <Checkbox
            checked={true}
            disabled={false}
            id="cars-select-item-123"
            onChange={[Function]}
          >
            Ford
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={false}
            id="cars-favourite-item-123"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="checkboxContainer col-md-6"
        key="234"
      >
        <div>
          <Checkbox
            checked={false}
            disabled={false}
            id="cars-select-item-234"
            onChange={[Function]}
          >
            Toyota
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={false}
            id="cars-favourite-item-234"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="checkboxContainer col-md-6"
        key="456"
      >
        <div>
          <Checkbox
            checked={true}
            disabled={true}
            id="cars-select-item-456"
            onChange={[Function]}
          >
            Mazda
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={true}
            id="cars-favourite-item-456"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="checkboxContainer col-md-6"
        key="645"
      >
        <div>
          <Checkbox
            checked={true}
            disabled={false}
            id="cars-select-item-645"
            onChange={[Function]}
          >
            McLaren
          </Checkbox>
        </div>
        <div
          className="favourite"
        >
          <FavouriteStarButton
            enabled={false}
            id="cars-favourite-item-645"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</Modal>
`;
